Increase audio length limit majorly on desktop

SomeGuyWhoLovesCoding opened the pull request:
So, there was an int64 conversion in the way of making ``time``, ``length``, ``dataLength``, and ``samples`` floats, causing an overflow once you get past the int64 limit. To fix that, I simply converted it to a float, and now you can load audio files over 6.6 hours with 44.1k sample rate!

I did make ``project/src/media/containers/OGG.cpp`` readable too.

I'm also planning to fix the memory leak problem and another problem with audio buffer when streaming audio. But I don't even know how I would fix that because I simply have never heard of some way to fix it entirely when doing ``new AudioSource(AudioBuffer.fromVorbisFile(VorbisFile.fromFile(sourceFile)))`` for the audio source (Which streams the ogg audio).

SomeGuyWhoLovesCoding committed:
Increase audio length limit majorly on desktop

SomeGuyWhoLovesCoding committed:
Hopefully this fixes the memory leak issue

SomeGuyWhoLovesCoding committed:
Oops

SomeGuyWhoLovesCoding committed:
Clean up a bit

SomeGuyWhoLovesCoding committed:
Update NativeAudioSource.hx

SomeGuyWhoLovesCoding committed:
Clean up inlines and stuff

joshtynjala commented on src/lime/_internal/backend/native/NativeAudioSource.hx:
```diff
@@ -199,47 +149,35 @@ class NativeAudioSource
 		AL.sourcePause(handle);
 
 		if (streamTimer != null)
-		{
```
Please don't remove braces like this. It makes it more difficult to filter through the PR to find the truly relevant parts. Additionally, though, while you may personally prefer to omit braces, others prefer to always include braces, even when optional, so best not to change what may have been an intentional choice.

SomeGuyWhoLovesCoding replied:
Ok

SomeGuyWhoLovesCoding committed:
Fix compiling error by using long long instead of float

C++ is kind of confusing when you write the code for the first time, but if you've learned haxe then yeah

SomeGuyWhoLovesCoding committed:
Update NativeAudioSource.hx

SomeGuyWhoLovesCoding committed:
Fix streaming on higher pitches

SomeGuyWhoLovesCoding committed:
Oops

SomeGuyWhoLovesCoding committed:
A few and satisfying audio stream bugfixes

I do think ``STREAM_TIMER_FREQUENCY`` is absolutely needed, and I think 20 is enough for it because it's actually the best frequency yet.
I also wish I could fix the constant memory allocation issue but had no luck.
Looping is untested with those bugfixes due to how it's implemented. I think I'm gonna rewrite it entirely, and make it rely on the current time.

SomeGuyWhoLovesCoding committed:
Update AudioSource.hx

SomeGuyWhoLovesCoding committed:
Only apply the fix when ``stream`` is true

SomeGuyWhoLovesCoding committed:
Merge branch 'develop' of https://github.com/SomeGuyWhoLovesCoding/lime into develop

SomeGuyWhoLovesCoding committed:
Update NativeAudioSource.hx

SomeGuyWhoLovesCoding committed:
Update NativeAudioSource.hx

SomeGuyWhoLovesCoding committed:
Update NativeAudioSource.hx

SomeGuyWhoLovesCoding commented:
The reason why I closed this because it fucks up the audio playback on some native targets, especially on hl
